test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/PropertyTrackingTests/DefaultChangeTrackingTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="DefaultChangeTrackingTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Batch;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.E2E.TestCommon;
+using Microsoft.OData.E2E.TestCommon.Common.Client.Default.Default;
+using Microsoft.OData.E2E.TestCommon.Common.Server.Default;
+using Microsoft.OData.E2E.TestCommon.Common.Server.PropertyTrackingTests;
+using Xunit;
+using Account = Microsoft.OData.E2E.TestCommon.Common.Client.Default.Account;
+using Customer = Microsoft.OData.E2E.TestCommon.Common.Client.Default.Customer;
+using HomeAddress = Microsoft.OData.E2E.TestCommon.Common.Client.Default.HomeAddress;
+using Order = Microsoft.OData.E2E.TestCommon.Common.Client.Default.Order;
+using Person = Microsoft.OData.E2E.TestCommon.Common.Client.Default.Person;
+
+namespace Microsoft.OData.Client.E2E.Tests.PropertyTrackingTests;
+
+/// <summary>
+/// End-to-end tests for default change tracking.
+/// </summary>
+public class DefaultChangeTrackingTests : EndToEndTestBase<DefaultChangeTrackingTests.TestsStartup>
+{
+    private readonly Uri _baseUri;
+
+    public class TestsStartup : TestStartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.ConfigureControllers(typeof(DefaultChangeTrackingTestsController), typeof(MetadataController));
+
+            services.AddControllers().AddOData(opt =>
+                opt.EnableQueryFeatures().AddRouteComponents("odata", DefaultEdmModel.GetEdmModel(), batchHandler: new DefaultODataBatchHandler()));
+        }
+    }
+
+    public DefaultChangeTrackingTests(TestWebApplicationFactory<TestsStartup> fixture) : base(fixture)
+    {
+        if (Client.BaseAddress == null)
+        {
+            throw new ArgumentNullException(nameof(Client.BaseAddress), "Base address cannot be null");
+        }
+
+        _baseUri = new Uri(Client.BaseAddress, "odata/");
+    }
+
+    // This test verifies that primitive and collection properties of entities can be updated using PATCH requests.
+    // It covers updating properties of various entity types, including inherited and complex types, as well as
+    // collection navigation properties and single value navigation properties.
+    [Fact]
+    public void UpdatePrimitiveAndCollectionPropertiesUsingPatch()

```

Comment:
Can this test be broken into 2 tests.. One for collection of primitives and the other collection of entity types for example.. or what the scenario is.